service/src/main/java/com/epam/esm/services/GiftCertificateService.java

```diff
@@ -0,0 +1,34 @@
+package com.epam.esm.services;
+
+import com.epam.esm.exceptions.ApplicationException;
+import com.epam.esm.exceptions.DataNotFoundException;
+import com.epam.esm.exceptions.WrongParameterException;
+import com.epam.esm.model.GiftCertificate;
+import org.springframework.data.domain.Pageable;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GiftCertificateService {
+    GiftCertificate saveGiftCertificate(GiftCertificate giftCertificate)
+            throws WrongParameterException, ApplicationException;
```

Comment:
There is **throws** method signature everywhere. Using Unchecked expetion can help to get rid of them.